src/services/phone-provider.ts

```diff
@@ -1,38 +1,36 @@
 import twilio from "twilio"
 
 import { baseLogger } from "@services/logger"
+import { getTwilioConfig } from "@config/app"
+import { TwilioError } from "@domain/errors"
 
-const getTwilioClient = () => {
-  const accountSid = process.env.TWILIO_ACCOUNT_SID
-  const apiKey = process.env.TWILIO_API_KEY
-  const apiSecret = process.env.TWILIO_API_SECRET
-
-  const client = twilio(apiKey, apiSecret, { accountSid })
-  return client
-}
+const client = twilio(getTwilioConfig().apiKey, getTwilioConfig().apiSecret, {
+  accountSid: getTwilioConfig().accountSid,
+})
 
 export const sendTwilioText = async ({ body, to, logger }) => {
-  const twilioPhoneNumber = process.env.TWILIO_PHONE_NUMBER
-  const provider = "twilio"
+  const twilioPhoneNumber = getTwilioConfig().twilioPhoneNumber
   try {
-    await getTwilioClient().messages.create({
+    await client.messages.create({
       from: twilioPhoneNumber,
       to,
       body,
     })
   } catch (err) {
-    logger.error({ err, provider }, "impossible to send text")
+    logger.error({ err }, "impossible to send text")
     return false
   }
 
-  logger.info({ to, provider }, "sent text successfully")
+  logger.info({ to }, "sent text successfully")
   return true
 }
 
 export const getCarrier = async (phone: string) => {
-  const result = await getTwilioClient()
-    .lookups.phoneNumbers(phone)
-    .fetch({ type: ["carrier"] })
-  baseLogger.info({ result }, "result carrier info")
-  return result
+  try {
```

Comment:
this service/methods should be refactored as Arvin suggested. We can even go further and make it part of notification service (open to discussion)